### PR TITLE
Add role="search" to global search

### PR DESCRIPTION
**Pull request by anselmbradford**
Includes search SVG updates from https://github.com/cfpb/cfgov-refresh/pull/3887

Fixes [GHE]/CFGOV/platform/issues/2588 (missing role="search" and other invalid HTML issues).
Fixes part of [GHE]/CFGOV/platform/issues/2310 (retains X at desktop).

## Additions

- Add `role="search"` to global-search.

## Changes

- Update to use SVG icons.
- Add content to empty labels.
- Add content to empty tab-trigger button.

## Removals

- Removes "clear" text from search and related tests.

## Testing

1. Pull this branch down and `gulp clean && gulp build`
1. Using the [WAVE](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh?hl=en-US) browser extension, check the homepage. There should be no errors in the search and it should show the search role.

## Screenshots

screen shot 2018-05-11 at 5 56 39 pm

## Todos

- This PR needs to be on hold till the OAH rate checker flag is switched in production.

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/molecules/global-search.html [hunk 1]
```diff
@@ -87,6 +97,13 @@
                 </ul>
             </div>
         </form>
-        <button class="m-global-search_tab-trigger" aria-hidden="true"></button>
+        <button class="m-global-search_tab-trigger u-visually-hidden" aria-hidden="true">
+          .
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/global-search.html]: If this didn't work for the main menu, am I safe to assume it doesn't work here either?

Comment (anselmbradford) [reply cfgov/jinja2/v1/_includes/molecules/global-search.html]: Yes, this is being covered in https://github.com/cfpb/cfgov-refresh/pull/4129

Comment (jimmynotjim) [reply cfgov/jinja2/v1/_includes/molecules/global-search.html]: Update this to `Collapse` and I think this PR will be good to go.

Comment (anselmbradford) [reply cfgov/jinja2/v1/_includes/molecules/global-search.html]: Fixed

**Pull request merged**